File: java/unidad_02/Problema_2_7/src/Problema_2_7.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * 
 * El programa, dados como datos tres números enteros, determina si
 * los mismos están en orden creciente.
 *
 * @author Angel P.
 * 
 */

public class Problema_2_7 {

    public static boolean ascendente(int[] numeros, int dimension) {

        for (int i = 0; i < dimension - 1; i++) {

            if (numeros[i] > numeros[i + 1]) {

                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {

        int dimension = 3;

        int[] numeros = new int[dimension];

        try (Scanner scanner = new Scanner(System.in)) {

            for (int i = 0; i < dimension; i++) {

                System.out.printf("Número %d: ", i + 1);
                numeros[i] = scanner.nextInt();
            }
        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        boolean sentencia = ascendente(numeros, dimension);

        if (!sentencia) {

            System.out.println("La lista de números no es ascendente");
            return;
        }
        System.out.println("La lista de números es ascendente");
    }
}

File: python/unidad_02/problema_2_7.py
"""
El programa, dados como datos tres números enteros, determina si
los mismos están en orden creciente.
"""

def ascendente(numeros, dimension):

    for i in range(dimension - 1):

        if numeros[i] > numeros[i + 1]:

            return False
    
    return True

def main():
    
    try: 

        dimension = 3

        numeros = []

        for i in range(dimension):

            numero = int(input(f"Número {i + 1}: "))
            numeros.append(numero)

    except ValueError as e:

        print(e)
        return
    
    sentencia = ascendente(numeros, dimension)

    if not sentencia:

        print("La lista de números no es ascendente")
        return
        
    print("La lista de números es ascendente")      

if __name__ == '__main__':

    main()